cpp/src/join/join_kernels.cuh

```diff
@@ -316,16 +320,16 @@ __global__ void probe_hash_table(multimap_type multi_map,
   if (probe_row_index < probe_table_num_rows) {
     const auto unused_key = multi_map.get_unused_key();
     const auto end        = multi_map.end();
-    auto found            = end;
 
     // Search the hash map for the hash value of the probe row using the row's
     // hash value to determine the location where to search for the row in the hash map
+    auto const probe_row_hash_value = [&]() {
+      auto const h = hash_probe(probe_row_index);
+      // Remap hash values equal to the empty sentinel value.
+      return (h == unused_key) ? (h - 1) : h;
```

Comment:
This lambda is repeated twice. Once in compute_output_size and once here. Also, the `return` line is repeated three times: those two and when building the hash table. So perhaps the remapping logic should be factored out to prevent future bugs.

Reply:
Done.